File: VB/DXSample/ViewModels/MainViewModel.vb
Imports DevExpress.Mvvm
Imports System
Imports System.Collections.ObjectModel

Namespace DXSample.ViewModels
	Public Class MainViewModel
		Inherits ViewModelBase

		Public Property Panels() As ObservableCollection(Of PanelViewModel)
			Get
				Return GetValue(Of ObservableCollection(Of PanelViewModel))()
			End Get
			Set(ByVal value As ObservableCollection(Of PanelViewModel))
				SetValue(value)
			End Set
		End Property

		Public Sub New()
			Panels = New ObservableCollection(Of PanelViewModel)() From {
				New PanelViewModel() With {
					.Caption = "One",
					.Content = "A regular panel",
					.State = State.Regular
				},
				New PanelViewModel() With {
					.Caption = "Two",
					.Content = "A regular panel",
					.State = State.Regular
				},
				New PanelViewModel() With {
					.Caption = "Three",
					.Content = "A regular panel",
					.State = State.Regular
				},
				New PanelViewModel() With {
					.Caption = "Four",
					.Content = "A float panel",
					.State = State.Float
				},
				New PanelViewModel() With {
					.Caption = "Five",
					.Content = "A panel",
					.ParentName = "layoutGroup"
				}
			}
		End Sub
	End Class
	Public Enum State
		None
		Float
		Regular
	End Enum

	Public Class PanelViewModel
		Inherits ViewModelBase

		Public Property Caption() As String
			Get
				Return GetValue(Of String)()
			End Get
			Set(ByVal value As String)
				SetValue(value)
			End Set
		End Property
		Public Property Content() As String
			Get
				Return GetValue(Of String)()
			End Get
			Set(ByVal value As String)
				SetValue(value)
			End Set
		End Property
		Public Property State() As State
			Get
				Return GetValue(Of State)()
			End Get
			Set(ByVal value As State)
				SetValue(value)
			End Set
		End Property
		Public Property ParentName() As String
			Get
				Return GetValue(Of String)()
			End Get
			Set(ByVal value As String)
				SetValue(value)
			End Set
		End Property
	End Class
End Namespace

File: CS/DXSample/ViewModels/MainViewModel.cs
using DevExpress.Mvvm;
using System;
using System.Collections.ObjectModel;

namespace DXSample.ViewModels {
    public class MainViewModel : ViewModelBase {
        public ObservableCollection<PanelViewModel> Panels {
            get { return GetValue<ObservableCollection<PanelViewModel>>(); }
            set { SetValue(value); }
        }

        public MainViewModel() {
            Panels = new ObservableCollection<PanelViewModel>() {
                new PanelViewModel() { Caption = "One", Content = "A regular panel", State = State.Regular },
                new PanelViewModel() { Caption = "Two", Content = "A regular panel", State = State.Regular },
                new PanelViewModel() { Caption = "Three", Content = "A regular panel", State = State.Regular },
                new PanelViewModel() { Caption = "Four", Content = "A float panel", State = State.Float },
                new PanelViewModel() { Caption = "Five", Content = "A panel", ParentName = "layoutGroup" },

            };
        }
    }
    public enum State { None, Float, Regular }

    public class PanelViewModel : ViewModelBase {
        public string Caption {
            get { return GetValue<string>(); }
            set { SetValue(value); }
        }
        public string Content {
            get { return GetValue<string>(); }
            set { SetValue(value); }
        }
        public State State {
            get { return GetValue<State>(); }
            set { SetValue(value); }
        }
        public string ParentName {
            get { return GetValue<string>(); }
            set { SetValue(value); }
        }
    }
}